File: 24-10-2024_Modify_the_Array.py
class Solution:
    def modifyAndRearrangeArr (self, arr) : 
        for i in range(1,len(arr)):
            if (arr[i]==arr[i-1]):
                arr[i-1]=arr[i-1]*2
                arr[i]=0
        i=0
        while (i<len(arr)):
            if (arr[i]==0):
                arr.remove(0)
                arr.append(0)
            i=i+1
        return arr

File: 24-10-2024_Modify_the_Array.cpp
class Solution {
public:
    void modifyAndRearrangeArr(vector<int>& arr) {
        for (int i = 1; i < arr.size(); i++) {
            if (arr[i] == arr[i - 1]) {
                arr[i - 1] *= 2;
                arr[i] = 0;
            }
        }

        int count = 0;
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] != 0) {
                arr[count++] = arr[i];
            }
        }

        while (count < arr.size()) {
            arr[count++] = 0;
        }
    }
};
